src/elements/cartoon/cartoonDataTransformer.ts

```diff
@@ -1,40 +1,72 @@
-import type { Breakpoint } from "@guardian/src-foundations";
 import { undefinedDropdownValue } from "../../plugin/helpers/constants";
 import type { FieldNameToValueMap } from "../../plugin/helpers/fieldView";
 import type { Asset } from "../helpers/defaultTransform";
 import type { MainImageData } from "../helpers/types/Media";
 import type { TransformIn, TransformOut } from "../helpers/types/Transform";
 import type { cartoonFields } from "./CartoonSpec";
 
+type ViewportSize = "small" | "medium" | "large"; // This used to be called "breakpoint"
+
+type Image = {
+  mimeType: "image/jpeg" | "image/png" | "image/svg+xml";
+  file: string;
+  width: number | string;
+  height: number | string;
+};
+
+type Variant = {
+  viewportSize: ViewportSize;
+  images: Image[];
+};
+
+type Fields = {
+  variants: Variant[];
+  role?: string;
+  photographer?: string;
+  caption?: string;
+  source?: string;
+  displayCredit?: string;
+};
+
 export type Element = {
   elementType: string;
-  fields: Record<string, string | undefined>;
+  fields: Fields;
   assets: Asset[];
-  elements?: Element[];
 };
 
 export const transformElementIn: TransformIn<
   Element,
   ReturnType<typeof cartoonFields>
-> = ({ fields, elements }) => {
-  const { role, photographer, caption, source } = fields;
+> = ({ fields }) => {
+  const { role, photographer, caption, source, variants } = fields;
+
+  const getImages = (breakpoint: ViewportSize): MainImageData[] => {
+    const variant = variants.find(
+      (variant) => variant.viewportSize === breakpoint
+    );
 
-  const getImages = (breakpoint: Breakpoint): MainImageData[] => {
-    if (Array.isArray(elements)) {
-      return elements
-        .filter(
-          (element) =>
-            element.elementType === "image" &&
-            element.fields.breakpoint === breakpoint
-        )
-        .map((element) => {
-          return {
-            mediaId: element.fields.mediaId,
-            mediaApiUri: element.fields.mediaApiUri,
-            assets: element.assets,
-            caption: element.fields.caption,
-          };
-        });
+    if (variant && Array.isArray(variant.images)) {
+      const assets = variant.images.map((image) => {
+        return {
+          assetType: "image",
+          mimeType: image.mimeType,
+          url: image.file,
+          fields: {
+            isMaster: undefined, // TODO: how to calculate this?
```

Comment:
I'd say this is quite important
```suggestion
            isMaster: undefined, // FIXME: how to calculate this?
```

Reply:
Apologies, this was a little sloppy, I had intended to leave a proper comment in the PR but forgot. 

I'm keen to implement these missing fields in this PR if I can, I just don't know how.

So, for example, given this element:

```
{
    elementType: "cartoon",
    fields: {
        variants: [
            {
                viewportSize: "large",
                images: [
                    {
                        mimeType: "image/jpeg",
                        file: "https://api.media.test.dev-gutools.co.uk/images/12345",
                        width: 456,
                        height: 789
                    }
                ]
            }
        ]
    }
}
```

How do I deduce `isMaster`, `mediaId`, `mediaApiUri`?